src/org/zaproxy/zap/extension/ascanrules/TestCrossSiteScriptV2.java

```diff
@@ -133,7 +134,13 @@ public void scan(HttpMessage msg, String param, String value) {
 			boolean attackWorked = false;
             HttpMessage msg2 = getNewMsg();
 			setParameter(msg2, param, Constant.getEyeCatcher());
-            sendAndReceive(msg2);
+            try {
+    			sendAndReceive(msg2);
+        	} catch (InvalidRedirectLocationException|UnknownHostException e) {
+        		// Not an error, just means we probably attacked the redirect location

```

Comment:
Right, I noticed that doing the same, I think it should be changed too.
There's no need to try checking the response if there's no response, not harmful but unnecessary.